### PR TITLE
Shade jackson-core into azure-core

### PR DESCRIPTION
**Pull request by alzimmermsft**
This PR shades jackson-core into azure-core and updates build and linting to ignore the shaded code. Additionally, contains instructions on how to update the shaded code in the future (if needed).

### CODE REVIEW

File: sdk/core/azure-core/pom.xml [hunk 1]
```diff
@@ -410,5 +412,58 @@
...
+
+              <artifactSet>
+                <includes>
+                  <include>com.fasterxml.jackson.core:jackson-core</include>
+                </includes>
+              </artifactSet>
+
+              <relocations>
+                <relocation>
+                  <pattern>com.fasterxml.jackson.core</pattern>
```
Comment (lmolkova) [inline sdk/core/azure-core/pom.xml:459]: I believe we need to comply with [Jackson's Apache 2](https://github.com/FasterXML/jackson-core/blob/2.14/LICENSE) and add the license and notice files. 

this post suggests to do so in META-INF folder and also reflecting it in the docs
https://opensource.stackexchange.com/questions/10823/shading-dependencies-with-apache-license

**Review by lmolkova — COMMENTED** [review]
Is there a way to prevent someone from accidentally modifying shaded jackson files? If not having a readme mentioning that those are external and not to be changed would be great.

**Comment by kasobol-msft** [comment]
> Is there a way to prevent someone from accidentally modifying shaded jackson files? If not having a readme mentioning that those are external and not to be changed would be great.

I think the most common solution is to have generated src like this in separate source root like `sdk\core\azure-core\src\main\generated` (or so).
Other option would be to ship `azure-core-jackson-shaded.jar` and have separate project.

**Comment by alzimmermsft** [comment]
> > Is there a way to prevent someone from accidentally modifying shaded jackson files? If not having a readme mentioning that those are external and not to be changed would be great.
> 
> I think the most common solution is to have generated src like this in separate source root like `sdk\core\azure-core\src\main\generated` (or so). Other option would be to ship `azure-core-jackson-shaded.jar` and have separate project.

For simplicities sake while working on this branch I'll keep it shaded how it is. I'll file issues to follow-up on this and the licensing later.

**Pull request merged**